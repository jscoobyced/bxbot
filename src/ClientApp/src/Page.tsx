import * as React from 'react';
import { BrowserRouter, Route } from 'react-router-dom';
import { CandleChartPageHoc } from './components/charts/CandleChartPageHoc';
import { Home } from './components/main/Home';
import { Layout } from './components/main/Layout';
import { ContentService } from './services/ContentService';
import { AllContent, Language } from './services/Models';
import { HtmlUtils } from './utils/HtmlUtils';

export interface PageState {
    allContent: AllContent;
}

export class Page extends React.Component<{}, PageState> {

    private contentService: ContentService;

    public constructor(props: any) {
        super(props);
        this.contentService = new ContentService();
        this.state = { allContent: this.contentService.defaultContent() };
    }

    public componentDidMount() {
        const allContent = this.updateContent(Language.English);
        this.setState({ allContent });
    }

    public render() {
        const routes = <Layout allContent={this.state.allContent} >
            <Route exact path='/' component={() => <Home homeContent={this.state.allContent.homeContent} />} />
            <Route path='/candle' component={CandleChartPageHoc} />
        </Layout>;
        return <BrowserRouter children={routes} basename={HtmlUtils.baseUrl()} />;
    }

    private updateContent(language: Language): AllContent {
        return this.contentService.updateContent(language);
    }
}