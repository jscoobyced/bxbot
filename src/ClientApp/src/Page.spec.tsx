import { shallow } from 'enzyme';
import * as React from 'react';
import { Page } from './Page';

test('Page content is preserved', () => {
    const page = shallow(<Page />);
    expect(page).toMatchSnapshot();
});
