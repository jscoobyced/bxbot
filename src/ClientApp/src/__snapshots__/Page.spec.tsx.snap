// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page content is preserved 1`] = `
<BrowserRouter
  basename=""
>
  <Layout
    allContent={
      Object {
        "homeContent": Object {
          "article": "Opening of bxbot. This website is a demo of ReactJS and netcore 2.1, and includes a Google Chart based crypto-currency trend and simple analysis. This website is my training material and might not work as you expect. Use it for education purpose, not as a trading tool.",
          "articleTitle": "Latest news - 2018-10-14",
        },
        "navigationMenuContent": Object {
          "about": "About",
          "candleChart": "Candle Chart",
          "cryptoCurrency": "Crypto-currency",
          "home": "Home",
          "websiteName": "bxbot",
        },
      }
    }
  >
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/candle"
    />
  </Layout>
</BrowserRouter>
`;
